src/content/blog/ga_qas.md

```diff
@@ -0,0 +1,233 @@
+
+<figure>
+    <img src='/public/images/ga-qas/ga-qas2.png' style="margin: auto" alt='description' />
+    <figcaption>Fig 2. The general pipeline of GA-QAS</figcaption>
+</figure>
+
+A genetic algorithm (GA) is a heuristic algorithm based on a genetic combination process. GA processes include Fitness evaluation, Selection, Cross-over, and Mutation. We treat ansatz as an individual in the population (where gates are its genes).
+
+|        Process        |                                                                                                     Method                                                                                                      |
+| :-------------------: | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+| Population generation |                                                A $n$ ansatz is initialized from a pool gate (Clifford$+R_i+CR_i, i\in\{x,y,z\}$) with certain provided metadata                                                 |
+|  Fitness evaluation   |                                                                             Evaluate each ansatz by calling `fitness_func(ansatz)`                                                                              |
+|       Selection       |                                                    Get fitness value from the fitness evaluation; simply sort from high to low and take the first half part                                                     |
+|      Cross-over       | Divide two ansatz $qc_1$ and $qc_2$ into four parts $\{qc_{11}, qc_{12}, qc_{21}, qc_{22}\}$, then combine each two parts into two new ansatz $qc_1^*=qc_{11}\vert qc_{22}$ and $qc_2^{*}=qc_{12}\vert qc_{21}$ |
+|       Mutation        |                                              Each gate on ansatz has a small probability of mutating to another gate (with the same number of qubits) in the pool                                               |
+
+_**Tab 1**. Detail of each operation in GA-QAS._
+
+The overall process can be viewed in _Fig. 2_.
+
+In GA, some hyper-parameters need to be considered and defined before you run GA-QAS:
+
+```py
+from qoop.evolution.environment import EEnvironmentMetadata
+env_metadata = EEnvironmentMetadata(
+        num_qubits, # As its name
+        depth, # Ansatz depth you want
+        num_circuit, # Number of ansatz per generation
+        num_generation, # Number of generation/iteration for GA
+        prob_mutate # Mutation probability, usually as small as 0.01 (1%)
+)
+```
+
+Then, you need to create an `EEnvironment` object, the important parameter is `fitness_func`, which is the function name that we declared above:
+
+```py
+from qoop.evolution.environment import EEnvironment
+env = EEnvironment(
+     metadata = env_metadata,
+     fitness_func = fitnessW,
+)
+```
```

Comment:
I think it's simple enough to merge these two code cells. WDYT?

Reply:
I agree with you, I merge these two code cells into one.